Source/UI/Buttons/BlockButton.swift

```diff
@@ -54,7 +54,7 @@ class BlockButton: PillButton {
 
         let shouldBlock = !self.isSelected
 
-        func complete() {
+        @MainActor func complete() {
```

Comment:
I don't know why I had to do this

Reply:
I made `AppController.shared.hideProgress()` a `@MainActor` function, which means it has to be called on the main thread. So the compiler wants you to ensure that `hideProgress()` is called on the main thread either by making `complete()` always be called on the main thread (which is what you did here), or you could have manually dispatched the call to `hideProgress()` to the main actor like this:
```
Task {
    await MainActor.run {
        AppController.shared.hideProgress()
    }
}
```

Reply:
My initial worry was because after merging the main branch, I got that compilation issue so I don't know why you could make hideProgress() a @MainActor function but not update this complete() function in BlockButton and still have the app to compile successfully.
Maybe I missed something while merging, like a extension declared in a file not included in the project, I don't know. I checked the branch that commit that change and it didn't update the BlockButton function. I didn't pull that branch to check if it compiles though.

Reply:
@martindsq ah no I think that's an important caveat for the `@MainActor` annotation. @MainActor only guarantees that a function will be called from the main thread when the function is being called from a "structured concurrency" context. I think a structured concurrency context means that you are 
1) in a function marked `async` 
2) in a `@Sendable` closure (basically an `async` closure).
3) in a `Task {` closure

Sometimes the compiler will still complain about calling an @MainActor function even if you aren't in one of these contexts, but it isn't guaranteed to catch all cases.